feature-extensions/apigee/terraform/apigee-gateway-func.tf

```diff
@@ -0,0 +1,130 @@
+# Zip up function for deploy
+data "archive_file" "apigee-gateway-zip" {
+  type        = "zip"
+  source_dir = "${path.module}/../jazz-apigee-proxy"
+  output_path = "${path.module}/../jazz-apigee-proxy.zip"
+}
+
+# Create a role for the Lambda function to exec under
+resource "aws_iam_role" "apigee-lambda-role" {
+  name = "${var.env_prefix}-jazz-apigee-lambda"
+  path = "/jazz/${var.env_prefix}/system/roles/"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "sts:AssumeRole"
+      ],
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+# Attach a policy to the lambda function role that lets it exec any lambda funcs
+# in this region that were installed under this account
+resource "aws_iam_role_policy" "apigee-lambda-policy" {
+  name = "${var.env_prefix}-jazz-apigee-lambda"
+  role = "${aws_iam_role.apigee-lambda-role.id}"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "lambda:InvokeFunction",
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Effect": "Allow",
+      "Resource": "arn:aws:lambda:${var.region}:${var.jazz_aws_accountid}:*"
+    }
+  ]
+}
+EOF
+}
+
+# Deploy the function with the correct role.
+resource "aws_lambda_function" "jazz-apigee-proxy" {
+  filename         = "${data.archive_file.apigee-gateway-zip.output_path}"
+  function_name    = "${var.env_prefix}-jazz-apigee-proxy"
+  role             = "${aws_iam_role.apigee-lambda-role.arn}"
+  handler          = "jazz-apigee-proxy.handler"
+  source_code_hash = "${data.archive_file.apigee-gateway-zip.output_base64sha256}"
+  runtime          = "nodejs8.10"
```

Comment:
This isn't being deployed thru Jazz, it's being deployed by Terraform directly to Lambda, so the fact that it's using the node8 runtime shouldn't affect anything.

Might as well start the move anyway, node10 is about to hit LTS, 8 is what everyone should be on at this point.

Reply:
Regarding your first comment, is there a reason we aren't doing this through Jazz? In our process Jazz Admin is able manage all core services (make changes, monitor and all other things just like user services) through their dashboard. Since this will be an exception to the process, I was wondering the rationale behind this.

For second comment, I agree we should move to the latest and greatest at the earliest for all the right reasons. My comment was mostly about how Jazz would deploy it. For the migration, we have almost completed moving off of 4.3, so I believe we are in a good place to start moving to 8.1 for sure..

Reply:
@suryajak how are we ready to move to 8.1 when lot of our core services are running on 4.3 and have not been upgraded yet. am i missing anything?

Reply:
We might need to discuss this more then. My understanding from the last chat was that this function would be optionally deployed into an existing Jazz install after the fact using a standalone installer, and that Jazz admin would merely know if this Jazz installation had Apigee support enabled/installed or not.

If the Apigee gateway function must be managed by Jazz Admin in some way, my understanding of the problem is out of whack.